lib/backend/k8s/resources/networkpolicy.go

```diff
@@ -420,23 +428,30 @@ func (npw *networkPolicyWatcher) processNPEvents() {
 		// event needs to able to be passed back into a Watch client so that we can resume watching
 		// when a watch fails.  The watch client is expecting a comma separated list of resource
 		// versions in the format <CRD NP Revision>/<k8s NP Revision>.
+		var value interface{}
 		switch e.Type {
 		case api.WatchModified, api.WatchAdded:
 			value = e.New.Value
 		case api.WatchDeleted:
 			value = e.Old.Value
 		}
-		oma, ok := value.(metav1.ObjectMetaAccessor)
-		if !ok {
-			log.Error("Resource returned from watch does not implement the ObjectMetaAccessor interface")
-			return
-		}
-		if isCRDEvent {
-			npw.crdNPRev = oma.GetObjectMeta().GetResourceVersion()
-		} else {
-			npw.k8sNPRev = oma.GetObjectMeta().GetResourceVersion()
+
+		if value != nil {
```

Comment:
Should we be using the ok from the  `e, ok = <-npw.crdNPWatch.ResultChan()` lines above rather than checking the value?

Also, if we do get a nil value should we be sending that nil value up the stack or instead converting it to an error (terminated, closed by remote)